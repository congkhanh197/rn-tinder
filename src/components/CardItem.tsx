import React, {useState} from 'react';
import {Text, View, Image, TouchableOpacity} from 'react-native';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import styles from './styles/CardItem.styles';

const COLORS = {
  enable: 'green',
  disable: 'gray',
};

interface IProps {
  picture: string;
  info: {
    title: string;
    value: string;
    iconName: string;
  }[];
}

const CardItem: React.FC<IProps> = props => {
  const [chooseIndex, setChooseIndex] = useState(0);
  const {picture, info} = props;
  return (
    <View style={styles.container}>
      <View style={styles.header} />
      <View style={styles.borderAvatar} />
      <View style={styles.defaultAvatar}>
        <Image
          source={{
            uri:
              'https://i1.wp.com/www.mvhsoracle.com/wp-content/uploads/2018/08/default-avatar.jpg',
          }}
          style={styles.avatar}
        />
      </View>
      <View style={styles.avatarWrapper}>
        <Image source={{uri: picture}} style={styles.avatar} />
      </View>
      <View style={styles.infoTextWrapper}>
        <Text style={styles.titleText}>{info[chooseIndex].title}</Text>
        <Text style={styles.detailText}>{info[chooseIndex].value}</Text>
      </View>
      <View style={styles.chooseButtonWrapper}>
        {info.map((item, index) => (
          <TouchableOpacity
            style={[
              {
                borderTopColor:
                  chooseIndex === index ? COLORS.enable : 'transparent',
              },
              styles.button,
            ]}
            key={item.title}
            onPress={() => {
              setChooseIndex(index);
            }}>
            <FontAwesome
              name={item.iconName}
              size={30}
              color={chooseIndex === index ? COLORS.enable : COLORS.disable}
            />
          </TouchableOpacity>
        ))}
      </View>
    </View>
  );
};

export default CardItem;
