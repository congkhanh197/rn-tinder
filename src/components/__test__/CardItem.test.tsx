import React from 'react';
import CardItem, { IPeopleInfo } from '../CardItem';
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer';
describe('Test Card Item', () => {
  test('should render', () => {
    const props: IPeopleInfo = {
      picture:'test',
      info:[{
        title: "testTitle",
        value: 'testValue',
        iconName: 'testIcon'
      }]
    }
    const tree = renderer.create(<CardItem {...props}/>);
    expect(tree).toMatchSnapshot();
  });
  test('should change choose icon', () => {
    const props: IPeopleInfo = {
      picture:'test',
      info:[{
        title: "testTitle",
        value: 'testValue',
        iconName: 'testIcon'
      }]
    }
    const wrapper = shallow(<CardItem {...props}/>);
    wrapper.find('TouchableOpacity').simulate('click');
  })
});
