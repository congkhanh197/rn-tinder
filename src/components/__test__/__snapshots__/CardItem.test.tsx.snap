// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Card Item should render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 4,
      "marginHorizontal": 75,
      "marginTop": 20,
      "width": 600,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#f9f9f9",
        "borderBottomColor": "gray",
        "borderBottomWidth": 1,
        "height": 100,
        "width": 598,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "gray",
        "borderRadius": 55,
        "borderWidth": 1,
        "height": 110,
        "justifyContent": "center",
        "position": "absolute",
        "top": 25,
        "width": 110,
        "zIndex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 50,
        "height": 100,
        "justifyContent": "center",
        "overflow": "hidden",
        "position": "absolute",
        "top": 30,
        "width": 100,
        "zIndex": 2,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://i1.wp.com/www.mvhsoracle.com/wp-content/uploads/2018/08/default-avatar.jpg",
        }
      }
      style={
        Object {
          "borderRadius": 50,
          "height": 100,
          "width": 100,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 50,
        "height": 100,
        "justifyContent": "center",
        "overflow": "hidden",
        "position": "absolute",
        "top": 30,
        "width": 100,
        "zIndex": 3,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "test",
        }
      }
      style={
        Object {
          "borderRadius": 50,
          "height": 100,
          "width": 100,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 30,
        "marginTop": 60,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "gray",
          "fontSize": 20,
        }
      }
    >
      testTitle
    </Text>
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 25,
        }
      }
    >
      testValue
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "marginBottom": 10,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderTopColor": "green",
          "borderTopWidth": 2,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": 35,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "green",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ?
      </Text>
    </View>
  </View>
</View>
`;
